signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/view/SignInWithAppleButton.kt

```diff
@@ -51,20 +66,27 @@ class SignInWithAppleButton @JvmOverloads constructor(
         (background as? GradientDrawable)?.cornerRadius = cornerRadius
 
         if (icon != null) {
+            this.icon = DrawableCompat.wrap(icon)
+
             val iconVerticalOffset =
                 resources.getDimensionPixelOffset(R.dimen.sign_in_with_apple_button_textView_icon_verticalOffset)
 
-            icon.setBounds(
+            this.icon.setBounds(
                 0,
                 iconVerticalOffset,
-                icon.intrinsicWidth,
-                icon.intrinsicHeight + iconVerticalOffset
+                this.icon.intrinsicWidth,
+                this.icon.intrinsicHeight + iconVerticalOffset
             )
 
-            textView.setCompoundDrawablesRelative(icon, null, null, null)
+            textView.setCompoundDrawablesRelative(this.icon, null, null, null)
+        } else {
+            this.icon = null
+        }
```

Comment:
This if/else can be easier to keep working in the future if we do an assignment once and map:

```kotlin
this.icon = icon?.let {
    …
}
```